### DISCUSSION TITLE
ReferenceError: Buffer.from is not a function at toBufferLE

### DISCUSSION THREAD

**Discussion by theking200794** [phantom/docs#158] [Help & Support]
### Summary

Hello when i am calling this after connecting to the wallet using
`resp = window.solana.connect()` and getting publicKey
 ```
const instruction = solanaWeb3.SystemProgram.transfer({
            fromPubkey: resp.publicKey,
            toPubkey: destPubkey,
            lamports: solanaWeb3.LAMPORTS_PER_SOL * lamports, 
          });
```
in my frontend i am using simple javascript, I am getting an error 
Buffer.from is not a function at toBufferLE. I tried using CDN but got almost the same error as Buffer.from is not defined on line  
`lamports: solanaWeb3.LAMPORTS_PER_SOL * lamports,`
I included cdn of the buffer too
`  <script src="https://cdn.jsdelivr.net/npm/buffer@6.0.3/index.min.js"></script>`
  I also tried including `window.Buffer = Buffer;`
Also tried to pass lamports using `BigInt` and `new Uint8Array` but no success.
I am integrating phantom in frontend using JavaScript in my flask website for doing solana transaction with the help of phantom wallet. I am using web3 cdn for solanaWeb3 which is 
`<script src="https://unpkg.com/@solana/web3.js@latest/lib/index.iife.js"></script>`
Need help thx

### Steps to Reproduce

1- Connect to phantom wallet using
          `const resp = await window.solana.connect();`
2- Inside SystemProgram.transfer method will through an error like "buffer is not defined" on line 
  `lamports: solanaWeb3.LAMPORTS_PER_SOL * lamports, `
```
const instruction = solanaWeb3.SystemProgram.transfer({
            fromPubkey: resp.publicKey,
            toPubkey: destPubkey,
            lamports: solanaWeb3.LAMPORTS_PER_SOL * lamports, 
          }); 
```
I am integrating in frontend using JavaScript in my flask website. 

### Phantom Version

23.19.0

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hello @theking200794,

I would recommend reviewing the answer in this [Stack Exchange Question](https://solana.stackexchange.com/questions/212/uncaught-referenceerror-buffer-is-not-defined-using-phantom-wallet-solana-and). The issue seems similar.

The other thing I noticed is that using the same variable name on either side of the assignment statement may be causing issues.

**Comment by theking200794** [comment]
Thx for replying @adamdelphantom. My env is JS not react or node.js maybe that's why web3 is throwing exception like `Buffer.from is not a function`. I already tried [Stack Exchange Question](https://solana.stackexchange.com/questions/212/uncaught-referenceerror-buffer-is-not-defined-using-phantom-wallet-solana-and) but all answers are related to react or node. 
you are talking about `window.Buffer = Buffer` one for assigning same var on both sides? Actually, I added cdn for Buffer thats why i was thinking from global scope it will get `Buffer` and will assign it to the `window.Buffer`. I also tried removing `window.Buffer = Buffer` too but no success.
